parser_library/src/parsing/parser_error_listener.h

```diff
@@ -58,8 +58,7 @@ class parser_error_listener_base : public antlr4::ANTLRErrorListener
         antlr4::atn::ATNConfigSet* configs) override;
 
 protected:
-    virtual void add_parser_diagnostic(
-        range diagnostic_range, diagnostic_severity severity, std::string code, std::string message) = 0;
+    virtual void add_parser_diagnostic(const std::function<diagnostic_op(const range&)>& diag_op, const range r) = 0;
```

Comment:
in this case we should be fine with just `diagnostic_op (&diag_op)(const range&)`, right?

Reply:
Ok